### PR TITLE
Get some more detail on app build timings

### PR DESCRIPTION
**Pull request by esoergel**
## Summary
This adds some info to the `build_zip` section of the app build timings:

image

I also found a (presumably small) redundancy and removed that.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story

The changes here seem pretty safe, but app builds are big and scary, so I'd be interested in a second opinion on the safety of the change.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/hqmedia/tasks.py [hunk 1]
```diff
@@ -213,13 +221,18 @@ def create_files_for_ccz(build, build_profile_id, include_multimedia_files=True,
 
     # Don't rebuild the file if it is already there
     if not (os.path.isfile(fpath) and settings.SHARED_DRIVE_CONF.transfer_enabled):
-        files, errors, file_count = _build_ccz_files(build, build_profile_id, include_multimedia_files,
-                                                     include_index_files, download_id, compress_zip,
-                                                     filename, download_targeted_version)
-        file_cache = _zip_files_for_ccz(fpath, files, current_progress, file_progress, file_count, compression,
-                                        task)
+        with build.timing_context("_build_ccz_files"):
```
Comment (esoergel) [inline corehq/apps/hqmedia/tasks.py:224]: I don't love this, for two reasons - It adds clutter, and unlike `@time_method`, this won't noop if the parent timer is inactive.  I assume the overhead of these is small enough that that's not worth worrying about, but I'd still be interested in alternative approaches.  Here's what came to mind:

1. Restructure `create_files_for_ccz` and the associated helper functions as a class, and use `time_method`.  That'd solve both issues, but it's a big refactor that I'm not sure is justified. 
2. Make a `@time_function` decorator that can fetch a timing context from one of the function args.  I actually did this, but then each function needs access to the timing context, and most of these don't currently have it (when only their outer scope needs it at the moment)
3. Modify `TimingContext` in some way so as not to start if the root or parent timer wasn't started.  Again, not sure this saves any appreciable computation time.

File: corehq/apps/hqmedia/tasks.py [hunk 2]
```diff
@@ -126,10 +126,18 @@ def build_application_zip(include_multimedia_files, include_index_files, domain,
                           download_targeted_version=False):
     DownloadBase.set_progress(build_application_zip, 0, 100)
     app = get_app(domain, app_id)
-    fpath = create_files_for_ccz(app, build_profile_id, include_multimedia_files, include_index_files,
-                                 download_id, compress_zip, filename, download_targeted_version,
-                                 task=build_application_zip, expose_link=True)
-    _expose_download_link(fpath, filename, compress_zip, download_id)
```
Comment (orangejenny) [inline corehq/apps/hqmedia/tasks.py:132]: Non-blocking: I'd have kept this line and removed the `expose_link` param, which looks like it's never passed as True except for here, for the sake of having one fewer param.

File: corehq/apps/hqmedia/tasks.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqmedia/tasks.py:224]: > this won't noop if the parent timer is inactive

What does happen if the parent timer is inactive?

I agree this adds some clutter, but I can live with it. Definitely don't think restructuring `create_files_for_ccz` is necessary.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/hqmedia/tasks.py (hunk 1 above)
Comment (esoergel) [reply corehq/apps/hqmedia/tasks.py:224]: It doesn't look at the parent timer at all, so this calls `time.time()` on each `__enter__` and `__exit__` and saves it to the timer object.

File: corehq/apps/hqmedia/tasks.py (hunk 2 above)
Comment (esoergel) [reply corehq/apps/hqmedia/tasks.py:132]: `create_files_for_ccz` is actually only used in two places, and both of them set it to `True`, so I just dropped the param.

**Pull request merged**